packages/protocol/contracts/L1/libs/LibEthDepositing.sol

```diff
@@ -101,10 +101,19 @@ library LibEthDepositing {
             }
         }
 
-        // resize the length of depositsProcessed to j.
         assembly {
             mstore(depositsProcessed, j)
-            depositsRoot := keccak256(depositsProcessed, mul(j, 32))
         }
+        depositsRoot = hashDeposits(depositsProcessed);
```

Comment:
The previous hashing of depositsProcessed is buggy, but I think it can get fixed simply using:
```solidity
 depositsRoot := keccak256(add(depositsProcessed, 32), mul(j, 32))
```

Reply:
I think maybe the expected value in the test (`0x8117066d69ff650d78f0d7383a10cc802c2b8c0eedd932d70994252e2438c636`) is also calculated wrong. How is it calculated exactly, better to calculate the hash inside the test.

Reply:
There is a PR for the looping being done in assembly and keccak256 being done outside here Dani: https://github.com/taikoxyz/taiko-mono/pull/13782/files

Much less readable but saves some gas